storage/client/src/main/java/com/azure/storage/file/ShareClient.java

```diff
@@ -7,76 +7,90 @@
     }
 
-    public Mono<Response<ShareInfo>> create(Map<String, String> metadata, int quotaInGB) {
-        throw new UnsupportedOperationException();
+    public Response<ShareInfo> create() {
+        return create(null, null);
     }
 
-    public Mono<Response<ShareSnapshotInfo>> createSnapshot(Map<String, String> metadata) {
-        throw new UnsupportedOperationException();
+    public Response<ShareInfo> create(Map<String, String> metadata, Integer quotaInGB) {
+        return client.create(metadata, quotaInGB).block();
     }
 
-    public Mono<VoidResponse> delete(String shareSnapshot) {
-        throw new UnsupportedOperationException();
+    public Response<ShareSnapshotInfo> createSnapshot() {
+        return createSnapshot(null);
     }
 
-    public Mono<Response<ShareProperties>> getProperties(String shareSnapshot) {
-        throw new UnsupportedOperationException();
+    public Response<ShareSnapshotInfo> createSnapshot(Map<String, String> metadata) {
+        return client.createSnapshot(metadata).block();
     }
 
-    public Mono<Response<ShareInfo>> setQuota(int quotaInGB) {
-        throw new UnsupportedOperationException();
+    public VoidResponse delete() {
+        return delete(null);
     }
 
-    public Mono<Response<ShareInfo>> setMetadata(Map<String, String> metadata) {
-        throw new UnsupportedOperationException();
+    public VoidResponse delete(String shareSnapshot) {
+        return client.delete(shareSnapshot).block();
     }
 
-    public Flux<SignedIdentifier> listAccessPolicy() {
-        throw new UnsupportedOperationException();
+    public Response<ShareProperties> getProperties() {
+        return getProperties(null);
     }
-    public Mono<Response<ShareInfo>> setAccessPolicy(List<SignedIdentifier> permissions) {
-        throw new UnsupportedOperationException();
+
+    public Response<ShareProperties> getProperties(String shareSnapshot) {
+        return client.getProperties(shareSnapshot).block();
     }
 
-    public Mono<Response<ShareStatistics>> getStatistics() {
-        throw new UnsupportedOperationException();
+    public Response<ShareInfo> setQuota(int quotaInGB) {
+        return client.setQuota(quotaInGB).block();
+    }
+
+    public Response<ShareInfo> setMetadata(Map<String, String> metadata) {
+        return client.setMetadata(metadata).block();
+    }
+
+    public Iterable<SignedIdentifier> listAccessPolicy() {
+        return client.listAccessPolicy().toIterable();
+    }
+
+    public Response<ShareInfo> setAccessPolicy(List<SignedIdentifier> permissions) {
+        return client.setAccessPolicy(permissions).block();
+    }
+
+    public Response<ShareStatistics> getStatistics() {
+        return client.getStatistics().block();
     }
 
-    public Mono<Response<DirectoryClient>> createDirectory(String directoryName, Map<String, String> metadata) {
+    public Response<DirectoryAsyncClient> createDirectory(String directoryName, Map<String, String> metadata) {
```

Comment:
Why we return AsyncClient here?

Reply:
I'll change that to the right type. I copied and pasted wrong.